modules/axhal/src/arch/loongarch64/trap.S

```diff
@@ -0,0 +1,98 @@
+
+.macro RESTORE_REGS
+    csrrd   $t0, LA_CSR_PRMD
```

Comment:
It reads PRMD many times to determine if the exception is from the userspace.

Is it possible to generate instructions to be executed at compile time instead of at runtime through macros arguments like riscv does?